File: SDKs/ix-api.net/2.1.0/python/src/sdk/models/operations/member_joining_rules_list.py
from dataclasses import dataclass, field
from typing import Any,Enum,List,Optional
from dataclasses_json import dataclass_json
from sdk.models import shared


@dataclass
class MemberJoiningRulesListQueryParams:
    id: Optional[List[str]] = field(default=None, metadata={'query_param': { 'field_name': 'id', 'style': 'form', 'explode': False }})
    network_service: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'network_service', 'style': 'form', 'explode': True }})
    

@dataclass
class MemberJoiningRulesListRequest:
    query_params: MemberJoiningRulesListQueryParams = field(default=None)
    
class MemberJoiningRulesList400ApplicationJSONTypeEnum(str, Enum):
    HTTPS_ERRORS_IX_API_NET_V2_VALIDATION_ERROR_HTML = "https://errors.ix-api.net/v2/validation-error.html"


@dataclass_json
@dataclass
class MemberJoiningRulesList400ApplicationJSON:
    detail: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'detail' }})
    instance: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'instance' }})
    status: Optional[Any] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'status' }})
    title: Optional[Any] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'title' }})
    type: MemberJoiningRulesList400ApplicationJSONTypeEnum = field(default=None, metadata={'dataclasses_json': { 'field_name': 'type' }})
    
class MemberJoiningRulesList401ApplicationJSONTypeEnum(str, Enum):
    HTTPS_ERRORS_IX_API_NET_V2_AUTHENTICATION_ERROR_HTML = "https://errors.ix-api.net/v2/authentication-error.html"
    HTTPS_ERRORS_IX_API_NET_V2_SIGNATURE_EXPIRED_HTML = "https://errors.ix-api.net/v2/signature-expired.html"


@dataclass_json
@dataclass
class MemberJoiningRulesList401ApplicationJSON:
    detail: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'detail' }})
    instance: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'instance' }})
    status: Optional[Any] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'status' }})
    title: Optional[Any] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'title' }})
    type: MemberJoiningRulesList401ApplicationJSONTypeEnum = field(default=None, metadata={'dataclasses_json': { 'field_name': 'type' }})
    
class MemberJoiningRulesList403ApplicationJSONTypeEnum(str, Enum):
    HTTPS_ERRORS_IX_API_NET_V2_PERMISSION_DENIED_HTML = "https://errors.ix-api.net/v2/permission-denied.html"


@dataclass_json
@dataclass
class MemberJoiningRulesList403ApplicationJSON:
    detail: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'detail' }})
    instance: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'instance' }})
    status: Optional[Any] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'status' }})
    title: Optional[Any] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'title' }})
    type: MemberJoiningRulesList403ApplicationJSONTypeEnum = field(default=None, metadata={'dataclasses_json': { 'field_name': 'type' }})
    

@dataclass
class MemberJoiningRulesListResponse:
    content_type: str = field(default=None)
    member_joining_rules: Optional[List[Any]] = field(default=None)
    status_code: int = field(default=None)
    member_joining_rules_list_400_application_json_object: Optional[MemberJoiningRulesList400ApplicationJSON] = field(default=None)
    member_joining_rules_list_401_application_json_object: Optional[MemberJoiningRulesList401ApplicationJSON] = field(default=None)
    member_joining_rules_list_403_application_json_object: Optional[MemberJoiningRulesList403ApplicationJSON] = field(default=None)
    

File: SDKs/ix-api.net/2.1.0/go/pkg/models/operations/memberjoiningruleslist.go
package operations

type MemberJoiningRulesListQueryParams struct {
	ID             []string `queryParam:"style=form,explode=false,name=id"`
	NetworkService *string  `queryParam:"style=form,explode=true,name=network_service"`
}

type MemberJoiningRulesListRequest struct {
	QueryParams MemberJoiningRulesListQueryParams
}

type MemberJoiningRulesList400ApplicationJSONTypeEnum string

const (
	MemberJoiningRulesList400ApplicationJSONTypeEnumHTTPSErrorsIxAPINetV2ValidationErrorHTML MemberJoiningRulesList400ApplicationJSONTypeEnum = "https://errors.ix-api.net/v2/validation-error.html"
)

type MemberJoiningRulesList400ApplicationJSON struct {
	Detail   *string                                          `json:"detail,omitempty"`
	Instance *string                                          `json:"instance,omitempty"`
	Status   *interface{}                                     `json:"status,omitempty"`
	Title    *interface{}                                     `json:"title,omitempty"`
	Type     MemberJoiningRulesList400ApplicationJSONTypeEnum `json:"type"`
}

type MemberJoiningRulesList401ApplicationJSONTypeEnum string

const (
	MemberJoiningRulesList401ApplicationJSONTypeEnumHTTPSErrorsIxAPINetV2AuthenticationErrorHTML MemberJoiningRulesList401ApplicationJSONTypeEnum = "https://errors.ix-api.net/v2/authentication-error.html"
	MemberJoiningRulesList401ApplicationJSONTypeEnumHTTPSErrorsIxAPINetV2SignatureExpiredHTML    MemberJoiningRulesList401ApplicationJSONTypeEnum = "https://errors.ix-api.net/v2/signature-expired.html"
)

type MemberJoiningRulesList401ApplicationJSON struct {
	Detail   *string                                          `json:"detail,omitempty"`
	Instance *string                                          `json:"instance,omitempty"`
	Status   *interface{}                                     `json:"status,omitempty"`
	Title    *interface{}                                     `json:"title,omitempty"`
	Type     MemberJoiningRulesList401ApplicationJSONTypeEnum `json:"type"`
}

type MemberJoiningRulesList403ApplicationJSONTypeEnum string

const (
	MemberJoiningRulesList403ApplicationJSONTypeEnumHTTPSErrorsIxAPINetV2PermissionDeniedHTML MemberJoiningRulesList403ApplicationJSONTypeEnum = "https://errors.ix-api.net/v2/permission-denied.html"
)

type MemberJoiningRulesList403ApplicationJSON struct {
	Detail   *string                                          `json:"detail,omitempty"`
	Instance *string                                          `json:"instance,omitempty"`
	Status   *interface{}                                     `json:"status,omitempty"`
	Title    *interface{}                                     `json:"title,omitempty"`
	Type     MemberJoiningRulesList403ApplicationJSONTypeEnum `json:"type"`
}

type MemberJoiningRulesListResponse struct {
	ContentType                                    string
	MemberJoiningRules                             []interface{}
	StatusCode                                     int64
	MemberJoiningRulesList400ApplicationJSONObject *MemberJoiningRulesList400ApplicationJSON
	MemberJoiningRulesList401ApplicationJSONObject *MemberJoiningRulesList401ApplicationJSON
	MemberJoiningRulesList403ApplicationJSONObject *MemberJoiningRulesList403ApplicationJSON
}
